### PR TITLE
Updating tracking data posted to Contributions service

### PR DESCRIPTION
**Pull request by andre1050**
## What does this change?
Updates the data inside the `SlotBodyEnd` component to be passed into the Contributions service in order to generate a proper query string for our first A/B test.

## Why?
So we can track clicks originating from the Default Epic served through our service - and do it in such a way that it's easy for the Contributions team to find them.

### CODE REVIEW

File: src/model/window-guardian.ts [hunk 1]
```diff
@@ -24,6 +24,9 @@ export interface WindowGuardianConfig {
     };
     switches: { [key: string]: boolean };
     tests?: { [key: string]: string };
+    ophan?: {
+        pageViewId: string;
+    };
```
Comment (oliverlloyd) [inline src/model/window-guardian.ts]: Is `ophan.pageViewId` the same as `CAPI.pageId`?

File: src/web/components/SlotBodyEnd.tsx [hunk 2]
```diff
@@ -9,11 +9,19 @@ const wrapperMargins = css`
 export const SlotBodyEnd = () => {
     const endpointUrl = 'https://contributions.guardianapis.com/epic';
 
-    const trackingParams = {};
+    const trackingParams = {
+        ophanPageId: window.guardian.config.ophan?.pageViewId,
```
Comment (oliverlloyd) [inline src/web/components/SlotBodyEnd.tsx]: I'm not sure what's best here but this raises the question: should we pass in data via props or read it all from a global window object? If we want scalability I'd be inclined to lean on react and use props but I'm not sure what the goals are here?

Comment (nicl) [reply src/web/components/SlotBodyEnd.tsx]: I think there's an ordering dependency here on ophan-tracker-js, which may or may not get loaded early enough for this. cc @oliverlloyd and @gtrufitt . Probably we want to load Ophan as a priority script as a few things likely rely on `pageViewId` being available early?

Comment (andre1050) [reply src/web/components/SlotBodyEnd.tsx]: The Ophan ID we see here is set as a global variable (inside window.guardian...) in the <head> on DCR, to ensure it's set as early as possible and before the app is rehydrated.

Comment (andre1050) [reply src/web/components/SlotBodyEnd.tsx]: Basically - the Ophan Page is generated and added to the global scope independently of any Ophan scripts - https://github.com/guardian/dotcom-rendering/blob/b5e01f6d2ec98122a0ca443793bde4aa145ae6d3/src/web/server/htmlTemplate.ts#L117 - with it being placed in the `<head>` it should be reliable enough.

File: src/model/window-guardian.ts (hunk 1 above)
Comment (andre1050) [reply src/model/window-guardian.ts]: I don't think so. It's generated in the HTML document directly when the app is rendered on the server. I don't think we have anything in place to use this on the server yet.
https://github.com/guardian/dotcom-rendering/blob/b5e01f6d2ec98122a0ca443793bde4aa145ae6d3/src/web/server/htmlTemplate.ts#L117

File: src/web/components/SlotBodyEnd.tsx (hunk 2 above)
Comment (oliverlloyd) [reply src/web/components/SlotBodyEnd.tsx]: This all makes a lot more sense to me now. Thanks for clarifying. Happy to move forwards

Comment (nicl) [reply src/web/components/SlotBodyEnd.tsx]: Update: turns out we set this in `htmlTemplate` so the ophan-tracker-js library doesn't set it on DCR (so no race-condition).

**Pull request merged**